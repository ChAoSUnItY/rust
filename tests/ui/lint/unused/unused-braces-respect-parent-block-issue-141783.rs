//@ run-rustfix
//@ check-pass
#[allow(unreachable_code)]
#[warn(unused_braces)]

fn main() {
    return { return }; //~ WARN: unnecessary braces
    if return { return } { return } else { return }
    match return { return } {
        _ => { return }
    }
    while return { return } {}
}
