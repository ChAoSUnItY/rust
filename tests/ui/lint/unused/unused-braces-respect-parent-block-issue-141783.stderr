warning: unnecessary braces around `return` value
  --> $DIR/unused-braces-respect-parent-block-issue-141783.rs:7:12
   |
LL |     return { return };
   |            ^^      ^^
   |
note: the lint level is defined here
  --> $DIR/unused-braces-respect-parent-block-issue-141783.rs:4:8
   |
LL | #[warn(unused_braces)]
   |        ^^^^^^^^^^^^^
help: remove these braces
   |
LL -     return { return };
LL +     return return;
   |

warning: 1 warning emitted

